gravitee-am-test/specs/gateway/mfa.jest.spec.ts

```
@@ -0,0 +1,689 @@
+    const loginLocation = authResponse.headers['location'];
+
+    expect(loginLocation).toBeDefined();
+    expect(loginLocation).toContain(`${process.env.AM_GATEWAY_URL}/${domain.hrid}/login`);
+    expect(loginLocation).toContain(`client_id=${clientId}`);
+
+    return authResponse;
+}
+
+const login = async (authResponse, user, clientId) => {
+    const loginResult = await extractXsrfTokenAndActionResponse(authResponse);
+    return await performFormPost(loginResult.action, '', {
+            "X-XSRF-TOKEN": loginResult.token,
+            "username": user.username,
+            "password": "SomeP@ssw0rd",
+            "client_id": clientId
+        }, {
+            'Cookie': loginResult.headers['set-cookie'],
+            'Content-type': 'application/x-www-form-urlencoded'
+        }
+    ).expect(302);
+}
+
+const enrollSMSFactor = async (authorize, factor, domain) => {
+    const authResult = await extractXsrfTokenAndActionResponse(authorize);
+    const enrollMFA = await performPost(authResult.action, '', {
+            "factorId": factor.id,
+            "phone": "+4401234567890",
+            "user_mfa_enrollment": true,
+            "X-XSRF-TOKEN": authResult.token
+        },
+        {
+            'Cookie': authResult.headers['set-cookie'],
+            'Content-type': 'application/x-www-form-urlencoded'
+        }).expect(302);
+
+    expect(enrollMFA.headers['location']).toBeDefined();
+    expect(enrollMFA.headers['location']).toContain(`${process.env.AM_GATEWAY_URL}/${domain.hrid}/oauth/authorize`);
+
+    return enrollMFA;
+}
+
+const createSMTPResource = async (domain, accessToken) => {
+    const smtp = await createResource(domain.id, accessToken, {
+        "type": "smtp-am-resource",
+        "configuration": "{\"host\":\"localhost\",\"port\":5025,\"from\":\"admin@test.com\",\"protocol\":\"smtp\",\"authentication\":false,\"startTls\":false}",
+        "name": "FakeSmtp"
+    });
+
+    expect(smtp).toBeDefined();
```

This is tautological - for the purpose of this test, it doesn't add anything to assert that `smtp` is both defined and not null, as they are essentially the same thing. Also I would argue that you don't need to assert whether smtp is null or defined because you are then asserting on `smtp.id` which would fail were `smtp` null/undefined.

---

removed the unnecessary assertions.